src/components/Subscriber/Subscriber.tsx

```diff
@@ -0,0 +1,100 @@
+import Input from '@/components/Input';
+import MultiSelect from '@/components/MultiSelect';
+import Button from '@/components/Button';
+import { getTags } from '@/apis/tag';
+import { useEffect, useState } from 'react';
+import { Formik, Form } from 'formik';
+import { subscriberValidationSchema } from '@/validations/subscriber';
+
+const Subscriber = ({
+  onSubmit,
+  selectedOptions,
+  setSelectedOptions,
+  subscriber = null,
+}: {
+  onSubmit: any;
+  selectedOptions: any;
+  setSelectedOptions: any;
+  subscriber: any;
+}) => {
+  const [tags, setTags] = useState([]);
+
+  useEffect(() => {
+    const fetchTagLists = async () => {
+      setTags(await getTags());
+    };
+
+    fetchTagLists();
+  }, []);
+
+  const tagsList =
+    Array.isArray(tags.data) && tags.data.length > 0
+      ? tags.data.map((tag: any) => ({
+          value: tag.id,
+          label: tag.name,
+        }))
+      : [];
+
+  const initialValues = {
+    email: subscriber?.email ? subscriber.email : '',
+    name: subscriber?.name ? subscriber.name : '',
+    phone: subscriber?.phone ? subscriber.phone : '',
+  };
```

Comment:
I think a better structure would be to reduce the number of checks. We can define `initialValues` first with default values and in case of subscriber we can just use the spread operator.
```js

let initialValues = {
  email: '',
  name: '',
  phone: '',
};
if (subscriber) {
  initialValues = {...initialValues, ...subscriber}
}
```

Reply:
Thank you for this suggestion.